src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -135,52 +142,97 @@ private void ConnectToWorkspace(Workspace workspace)
                 ResetLastParsedDocument();
 
                 _workspace = workspace;
-                _workspace.WorkspaceChanged += this.OnWorkspaceChanged;
-                _workspace.DocumentOpened += this.OnDocumentOpened;
-                _workspace.DocumentActiveContextChanged += this.OnDocumentActiveContextChanged;
 
-                var textContainer = _subjectBuffer.AsTextContainer();
+                if (_workspace != null)
+                {
+                    _isClassificationOnlyWorkspace = false;
+                    _workspace.WorkspaceChanged += this.OnWorkspaceChanged;
+                    _workspace.DocumentOpened += this.OnDocumentOpened;
+                    _workspace.DocumentActiveContextChanged += this.OnDocumentActiveContextChanged;
+
+                    var textContainer = _subjectBuffer.AsTextContainer();
 
-                var documentId = _workspace.GetDocumentIdInCurrentContext(textContainer);
-                if (documentId != null)
+                    var documentId = _workspace.GetDocumentIdInCurrentContext(textContainer);
+                    if (documentId != null)
+                    {
+                        var document = workspace.CurrentSolution.GetDocument(documentId);
+                        if (document != null)
+                        {
+                            EnqueueParseSnapshotTask(document);
+                            return;
+                        }
+                    }
+                }
+                else
                 {
-                    var document = workspace.CurrentSolution.GetDocument(documentId);
-                    if (document != null)
+                    // This buffer isn't being tracked by any workspace. In Visual Studio, this
+                    // can happen if it is not in the Running Document Table. We'll try to do
+                    // basic syntax classification anyway.
+
+                    _isClassificationOnlyWorkspace = true;
+                    if (_languageName != null)
                     {
-                        EnqueueParseSnapshotTask(document);
+                        _workspace = new AdhocWorkspace();
+                        var solution = _workspace.CreateSolution(SolutionId.CreateNewId());
+                        var project = solution.AddProject(name: string.Empty, assemblyName: string.Empty, language: _languageName);
+
+                        var parsedSnapshot = _subjectBuffer.CurrentSnapshot;
+                        var document = project.AddDocument(name: string.Empty, text: _subjectBuffer.CurrentSnapshot.AsText());
```

Comment:
We never put this new solution back into `workspace`, so `workspace.CurrentSolution` will never have it.  I suppose that doesn't matter, because we only ever look at the new snapshot coming in and update `_lastParsedDocument`?